beacon-chain/powchain/log_processing.go

```diff
@@ -76,25 +76,23 @@ func (w *Web3Service) ProcessDepositLog(depositLog gethTypes.Log) {
 
 	// We then decode the deposit input in order to create a deposit object
 	// we can store in our persistent DB.
+	validData := true
 	depositInput, err := helpers.DecodeDepositInput(depositData)
 	if err != nil {
 		log.Errorf("Could not decode deposit input %v", err)
-		return
+		validData = false
 	}
 
 	deposit := &pb.Deposit{
 		DepositData:     depositData,
 		MerkleTreeIndex: index,
 	}
 
-	validData := true
-
 	// Make sure duplicates are rejected pre-chainstart.
-	if !w.chainStarted {
+	if !w.chainStarted && validData {
 		var pubkey = fmt.Sprintf("#%x", depositInput.Pubkey)
```

Comment:
`depositInput.Pubkey` will panic if there is an invalid deposit. Maybe add a mock pubkey  `[]byte("InvalidDeposit")` or something like that.

Reply:
This code block won't run in this case because we are checking if it is valid data beforehand

Reply:
ah true, yeah it wont